src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -261,7 +261,7 @@ def load_arguments(self, _):
         c.argument('nsg', help='The name to use when creating a new Network Security Group (default) or referencing an existing one. Can also reference an existing NSG by ID or specify "" for none.', arg_group='Network')
         c.argument('nsg_rule', help='NSG rule to create when creating a new NSG. Defaults to open ports for allowing RDP on Windows and allowing SSH on Linux.', arg_group='Network', arg_type=get_enum_type(['RDP', 'SSH']))
         c.argument('application_security_groups', min_api='2017-09-01', nargs='+', options_list=['--asgs'], help='Space-separated list of existing application security groups to associate with the VM.', arg_group='Network')
-
+        c.argument('workspace', arg_group='Monitor', help='Name or ID of Log Analytics Workspace. If name is given, the workspace should be in the same resource group with the vm, otherwise a new workspace will be created.')
```

Comment:
Another question for workspace for monitoring is when you need to create workspace for users, why not let user specify location, sku and other parameters like you do in CRUD command?

Reply:
the original design is to let user input an existed workspace instead of creating a new one. It's a vm command, so we only want to support day 0 scenario.

Reply:
To create a workspace, users need to use monitor command, right? Is it known for users?

Reply:
I don't think we should provide such information to user. For any commands which need vnet, we don't tell use to use "az network vnet create" to create a vnet. Log Analytics Workspace is a hint for user. If they don't know anything, we will create a default workspace for them.